### PR TITLE
Update for 2023

### PR DESCRIPTION
**Pull request by sebastian-berlin-wmse**
Based on changes made in project structure for 2023, including:
* Program information is now extracted from the project file rather
than a wikipage. Strategies are no longer included.
* Program overview page is no longer created.
* The projects year page now only has headers for programs and
doesn't include the numbers.
* The config used by WMSE is now included as default.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,15 +2,16 @@ This bot adds initial information at the start of year for Wikimedia Sverige. It
 
 # Installation
 Install required libraries with pip:
-
-`$ pip install -r requirements.txt`
+```
```
Comment (lokal-profil) [inline README.md:5]: I prefer this implementation to that in #46. Could possibly add the `console` language tag though for additional context

File: config.yaml [hunk 2]
```diff
@@ -63,38 +63,25 @@ wiki:
...
                 Föreningen <YEAR>: Föreningen
                 Projekt <YEAR>: Projekt efter verksamhetsår
                 Ledningsrapporter <YEAR>: Ledningsrapporter
                 Ansökningar <YEAR>: Ansökningar
+                Omnämnande <YEAR>:
+                    - Omnämnande
+                    - Media
                 Resor <YEAR>: Resor
                 Årsmöte <YEAR>: Årsmöte
                 Checklistor <YEAR>: Checklistor
```
Comment (lokal-profil) [inline config.yaml:75]: Would we not expect this,  or a replacement,  in 2024? Its included on [Mall:Metrics-sida](se.wikimedia.org/wiki/Mall:Metrics-sida)

**Review by lokal-profil — COMMENTED** [review]
Wiki.py I still need to check on a proper screen

**Comment by lokal-profil** [comment]
Should the Readme explivitly list the expected .env variables? 

One is implicitly listed but it is not clear if it is the only one

File: wiki.py [hunk 3]
```diff
@@ -494,163 +483,11 @@ def add_project(self, number, swedish_name, english_name):
```
Comment (lokal-profil) [inline wiki.py:485]: New param should be documented in docstring

**Review by lokal-profil — CHANGES_REQUESTED** [review]
I've looked through the code and left some comments. haven't tried to spin it up yet though

File: config.yaml (hunk 2 above)
Comment (sebastian-berlin-wmse) [reply config.yaml:75]: I don't understand what you mean. From https://phabricator.wikimedia.org/T336757 it sounds like we should not add these.

**Comment by sebastian-berlin-wmse** [comment]
> Should the Readme explivitly list the expected .env variables?
> 
> One is implicitly listed but it is not clear if it is the only one

I guess that it may be helpful to list env variables if there are a lot of them. Now there's only one and it's mentioned where it's relevant.

File: config.yaml (hunk 2 above)
Comment (lokal-profil) [reply config.yaml:75]: I think the comment somehow got attached to the wrong row (looks right in file view but not on the PR view).

My comment is about the removal of `Globala mätetal <YEAR>: Globala mätetal`

**Comment by lokal-profil** [comment]
> > Should the Readme explivitly list the expected .env variables?
> > One is implicitly listed but it is not clear if it is the only one
> 
> I guess that it may be helpful to list env variables if there are a lot of them. Now there's only one and it's mentioned where it's relevant.

Fair enough.

Does the program still work fine if the env variable is provided directly but the `.env` file is missing?
Does it fail gracefully (or at least in an informative way) if the env variable cannot be found? Since it's more likely to be missing now than when it was in the config.

**Comment by sebastian-berlin-wmse** [comment]
> Does the program still work fine if the env variable is provided directly but the `.env` file is missing?

Yes. You can use either. If you have both the value from the command line take precedence.

> Does it fail gracefully (or at least in an informative way) if the env variable cannot be found? Since it's more likely to be missing now than when it was in the config.

It raises an exception with the error from Phab's API. It looks like this:

```
[...]
phab.PhabApiError: Error from Phabricator API: Session key is not present.
CRITICAL: Exiting due to uncaught exception <class 'phab.PhabApiError'>
2023-06-19 16:18:35,715 [CRITICAL] (logging): Exiting due to uncaught exception <class 'phab.PhabApiError'>
```

**Comment by lokal-profil** [comment]
@sebastian-berlin-wmse the question about the config.yml is still unresolved

>My comment is about the removal of `Globala mätetal <YEAR>: Globala mätetal`

File: config.yaml (hunk 2 above)
Comment (sebastian-berlin-wmse) [reply config.yaml:75]: Should it still be called "Globala mätetal" or should it also change name?

Comment (lokal-profil) [reply config.yaml:75]: I think we can keep the name. It's an internal category and it works for both Global Metrics and Core metrics (they are both types of global metrics)

Comment (lokal-profil) [reply config.yaml:75]: I re-added it in the last commit since the removal did not sound intentional

**Review by lokal-profil — APPROVED** [review]

**Comment by lokal-profil** [comment]
@sebastian-berlin-wmse Do you want to check that this still works with #48 (as per your comments there). After that (and the resulting rebase) it should be good to merge.

**Pull request merged**